src/MSBuild/XMake.cs

```diff
@@ -1292,6 +1303,25 @@ string outputResultsCache
             return success;
         }
 
+        private static bool PrintTargets(string projectFile, string toolsVersion, Dictionary<string, string> globalProperties, TextWriter targetsWriter, ProjectCollection projectCollection)
+        {
+            try
+            {
+                Project project = projectCollection.LoadProject(projectFile, globalProperties, toolsVersion);
+
+                targetsWriter.WriteLine(string.Join(Environment.NewLine, project.Targets.Keys));
```

Comment:
Can you call this in a loop?
```suggestion
                foreach (string target in project.Targets.Keys)
                {
                    targetsWriter.WriteLine(target);
                }
```

That way we don't have to allocate a big string? (it's not super critical here, but it's good practice)